OSDOCS-5987: adds networking batch of bugs (#66489)
openshift/openshift-docs: release_notes/ocp-4-14-release-notes.adoc

* `segfault` failures that were related to HaProxy have been resolved. Users should no longer receive these errors. (link:https://issues.redhat.com/browse/OCPBUGS-11595[*OCPBUGS-11595*])

* Previously, CoreDNS terminated unexpectedly if a user created an `EndpointSlice` port without a port number. With this update, validation was added to CoreDNS to prevent it from unexpectedly terminating. (link:https://issues.redhat.com/browse/OCPBUGS-19805[*OCPBUGS-19805*])

> * Previously, the OpenShift router directed traffic to a route with a weight of `0 ` when it had only one back end. With this update, the router will not send traffic to routes with a single backend with weight `0`. (link:https://issues.redhat.com/browse/OCPBUGS-16623[*OCPBUGS-16623*])

Reviewer: Up to you: 

```suggestion
* Previously, the OpenShift router directed traffic to a route with a weight of `0 ` when it had only one back end. With this update, the router no longer sends traffic to routes with a single backend with weight `0`. (link:https://issues.redhat.com/browse/OCPBUGS-16623[*OCPBUGS-16623*])
```

Previously, the Ingress Operator created its canary route without specifying the `spec.subdomain` or the `spec.host` parameter on the route. Usually, this caused the API server to use the cluster's Ingress domain, which matches the domain of the default Ingress Controller, to set a default value for the `spec.host` parameter. However, if you configured the cluster by using the `appsDomain` option to set an alternative Ingress domain, the route host would have the alternative domain. Further, if you deleted the canary route, the route would be recreated with a domain that did not match the default Ingress Controller's domain, which would cause canary checks to fail. Now, the Ingress Controller specifies the `spec.subdomain` parameter when it creates the canary route. If you use the `appsDomain` option to configure your cluster and then delete the canary route, the canary checks do not fail. (link:https://issues.redhat.com/browse/OCPBUGS-16089[*OCPBUGS-16089*])

> Previously, the Ingress Operator did not check status of DNS records in public hosted zones when updating the Operator status.

Reviewer: ```suggestion
* Previously, the Ingress Operator did not check status of DNS records in public hosted zones when updating the Operator status.
```

This caused the Ingress Operator to report the DNS status as `ready` when there could be errors in DNS records in public hosted zones. Now, the Ingress Operator checks the status of both public and private hosted zones, which fixes the issue. (link:https://issues.redhat.com/browse/OCPBUGS-15978[*OCPBUGS-15978*])

> * Previously, the CoreDNS `bufsize` setting was configured as 512 bytes. Now, the maximum size of the buffer for {product-title} CoreDNS is 1232 bytes. This modification enhances DNS performance by reducing the occurrence of DNS truncations and retries. (link:https://issues.redhat.com/browse/OCPBUGS-15605[*OCPBUGS-15605*])

Reviewer: Is this a bug or an enhancement?

> * Previously, the Ingress Operator would specify the `spec.template.spec.hostNetwork: true` parameter on a router deployment without specifying the `spec.template.spec.containers[*].ports[*].hostPort`. This caused the API server to set a default value for each port's `hostPort` field, which the Ingress Operator would then detect as an external update and attempt to revert it. Now, the Ingress Operator no longer incorrectly performs these updates resolving the issue. (link:https://issues.redhat.com/browse/OCPBUGS-14995[*OCPBUGS-14995*])

Reviewer: ```suggestion
* Previously, the Ingress Operator would specify the `spec.template.spec.hostNetwork: true` parameter on a router deployment without specifying the `spec.template.spec.containers[*].ports[*].hostPort`. This caused the API server to set a default value for each port's `hostPort` field, which the Ingress Operator would then detect as an external update and attempt to revert it. Now, the Ingress Operator no longer incorrectly performs these updates. (link:https://issues.redhat.com/browse/OCPBUGS-14995[*OCPBUGS-14995*])
```

* Currently, users cannot modify the `interface-specific` safe sysctl list by updating the `cni-sysctl-allowlist` config map in the `openshift-multus` namespace. As a workaround, you can modify, either manually or with a DaemonSet, the `/etc/cni/tuning/allowlist.conf` file on the node or nodes. (link:https://issues.redhat.com/browse/OCPBUGS-11046[*OCPBUGS-11046*])

> * In {product-title} 4.14, all nodes use Linux control group version 2 (cgroup v2) for internal resource management in alignment with the default RHEL 9 configuration. However, if you apply a performance profile in your cluster, the low latency tuning features associated with the performance profile do not support cgroup v2.

Reviewer: ```suggestion
* In {product-title} 4.14, all nodes use Linux control group version 2 (cgroup v2) for internal resource management in alignment with the default {op-system-base} 9 configuration. However, if you apply a performance profile in your cluster, the low latency tuning features associated with the performance profile do not support cgroup v2.
```
